src/lib/operators/join_sort_merge.cpp

```diff
@@ -867,8 +895,8 @@ class JoinSortMerge::JoinSortMergeImpl : public AbstractReadOnlyOperatorImpl {
   * Executes the SortMergeJoin operator.
   **/
   std::shared_ptr<const Table> _on_execute() override {
-    bool include_null_left = (_mode == JoinMode::Left || _mode == JoinMode::FullOuter);
-    bool include_null_right = (_mode == JoinMode::Right || _mode == JoinMode::FullOuter);
+    const auto include_null_left = (_mode == JoinMode::Left || _mode == JoinMode::FullOuter);
+    const auto include_null_right = (_mode == JoinMode::Right || _mode == JoinMode::FullOuter);
```

Comment:
Why change the value type to auto? I think writing bool is more readable because you instantly know what you are dealing with instead of having to figure out the value type yourself.

Reply:
Matter of taste. We had this discussions a couple of times. Especially if it's not easy to determine what is the returned value. In this case, I'd say you need to read the right part of the equal sign anyways and it should be obvious. For other cases, it might be harder, but tbh I don't remember all of the discussion. But we settled for auto pretty much everywhere.